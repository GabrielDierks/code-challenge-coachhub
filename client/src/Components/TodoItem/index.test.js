import React from "react"
import Enzyme, { shallow } from "enzyme"
import { TodoItem } from "./index"
import renderer from "react-test-renderer"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() })

test("Todo Item snapshot", () => {
  const component = renderer.create(
    <TodoItem name="test" description="testdescription" checked={false} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test("Check a custom Todo", () => {
  const Todo = shallow(
    <TodoItem name="test" description="testdescription" checked={false} />
  )

  Todo.find("input").simulate("change")

  Todo.find("input").equals(true)
})
