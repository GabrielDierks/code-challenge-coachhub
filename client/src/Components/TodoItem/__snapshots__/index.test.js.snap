// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Item snapshot 1`] = `
<li>
  <label
    className="container"
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </label>
  <span
    className="title"
  >
    test
  </span>
  <button
    className="red"
    onClick={[Function]}
  />
  <button
    className="blue"
    onClick={[Function]}
  />
  <span
    className="text"
    maxLength="599"
    onChange={[Function]}
  >
    testdescription
  </span>
  <span
    className="text text-date"
  >
    last updated: 
    May 13th 2019, 19:15:55
  </span>
</li>
`;
